src/const_eval.md

```diff
@@ -23,7 +23,10 @@ to be run.
 * [Const parameters].
 * [Paths] to [functions] and [constants].
   Recursively defining constants is not allowed.
-* Paths to [statics]. These are only allowed within the initializer of a static.
+* Paths to immutable [statics] with these exception with these restrictions.
+    * Reads out of and writes into [`extern` statics] are not allowed.
+    * Reads out of and writes into either a `static` with data equipped with interior mutability,
+      or a whole or parts of `static mut`, are not allowed.
```

Comment:
I am confused by the wording here. Why does `static mut` emphasize this "whole or part of", but interior mutable `static` does not? The two are treated exactly the same so they should also be described the same.

The actual underlying check is simply "is this global mutable memory -- if yes, reject read", and "if this memory global -- if yes, reject write".

Reply:
Is this code accepted?
```rust
static FOO: (u8, UnsafeCell<u8>) = (0, UnsafeCell::new(1));

const BAR: u8 = FOO.0;
```